src/services/helpers.ts

```diff
@@ -117,3 +124,49 @@ export function addPagersFn<T>(
     next,
   };
 }
+
+export function addCustomPropertiesToConfig(
+  customProperties: string[],
+  httpInstance: HttpInstance,
+  requestOptions?: OptionsBase
+) {
+  // All custom properties normalization is done on the request and the response
+  // In requests keys are converted from camel to snake case
+  // In responses keys are converted from snake to camel case
+  // To avoid having to set all keys in the array as camel and as snake manually, each key
+  // is converted here to both camel and snake case before being put in the config.
+  const camilizedCustomProperties = customProperties.map(camelize);
+  const snakifiedCustomProperties = customProperties.map(decamelize);
+
+  // First check local request settings
+  // If turned off on the request, don't skip on any properties
+  if (requestOptions?.skipCaseNormalizationForCustomProperties === false) {
+    return requestOptions;
+  }
+
+  // If turned on, on the request, skip on the custom properties
+  if (requestOptions?.skipCaseNormalizationForCustomProperties === true) {
+    return {
+      ...requestOptions,
+      skipKeyNormalizationForProperties: [
+        ...camilizedCustomProperties,
+        ...snakifiedCustomProperties,
+      ],
+    };
+  }
+
+  // If nothing is set on the request, we need to check the global settings
+  // If globally turned off, don't skip on any properties
+  if (httpInstance.skipCaseNormalizationForCustomProperties === false) {
+    return requestOptions;
+  }
```

Comment:
(opinion/discussion) Could this be simplified if the toggle and the custom fields were different variables?

A `customProperties` variable  would always be added to the request object if applicable to the endpoint. (This would also aid the future RQL thingy we've discussed)

A `skipCaseNormalizationForCustomProperties` is also in the request object to toggle the feeding of the customProperties to decamalize/camelize or not. Which could be done at the last moment in `decamilizeRequestData` or `camelizeResponseData`

Keeping the definition there are "custom properties" in an endpoint and how that influences the behavior with extra toggles separate.

I might be missing stuff though, happy to discuss

Reply:
💯  love this idea